src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -338,16 +339,23 @@ private void SetTransferMode(TransferMode transferMode)
                         {
                             _logger.RaiseReceiveEvent(_connectionId);
 
-                            var receivedHandler = Received;
-                            if (receivedHandler != null)
+                            // Copying the callbacks to avoid concurrency issues
+                            ReceiveCallBack[] callBackCopies;
+                            lock (_callBacks)
+                            {
+                                callBackCopies = new ReceiveCallBack[_callBacks.Count];
+                                _callBacks.CopyTo(callBackCopies);
+                            }
+
+                            foreach (var callBackObject in callBackCopies)
                             {
                                 try
                                 {
-                                    await receivedHandler(buffer);
+                                    await callBackObject.InvokeAsync(buffer);
                                 }
                                 catch (Exception ex)
                                 {
-                                    _logger.ExceptionThrownFromHandler(_connectionId, nameof(Received), ex);
+                                    _logger.ExceptionThrownFromHandler(_connectionId, nameof(callBackObject), ex);
```

Comment:
Log message was about the 'name' event handler, something needs to change here.

Reply:
Message is now "An exception was thrown from the 'callBackObject' callback". I think 'OnReceived' is more appropriate. And the function name should be `ExceptionThrownFromCallback`